build.rs

```diff
@@ -147,14 +164,16 @@ fn build_sse2_sse41_avx2_assembly() {
     println!("cargo:rustc-cfg=blake3_sse41_ffi");
     println!("cargo:rustc-cfg=blake3_avx2_ffi");
     let mut build = new_build();
-    if is_windows_msvc() {
-        build.file("c/blake3_sse2_x86-64_windows_msvc.asm");
-        build.file("c/blake3_sse41_x86-64_windows_msvc.asm");
-        build.file("c/blake3_avx2_x86-64_windows_msvc.asm");
-    } else if is_windows_gnu() {
-        build.file("c/blake3_sse2_x86-64_windows_gnu.S");
-        build.file("c/blake3_sse41_x86-64_windows_gnu.S");
-        build.file("c/blake3_avx2_x86-64_windows_gnu.S");
+    if is_windows_target() {
+        if use_msvc_asm() {
+            build.file("c/blake3_sse2_x86-64_windows_msvc.asm");
+            build.file("c/blake3_sse41_x86-64_windows_msvc.asm");
+            build.file("c/blake3_avx2_x86-64_windows_msvc.asm");
+        } else {
+            build.file("c/blake3_sse2_x86-64_windows_gnu.S");
+            build.file("c/blake3_sse41_x86-64_windows_gnu.S");
+            build.file("c/blake3_avx2_x86-64_windows_gnu.S");
```

Comment:
@sneves are there any potential ABI mismatch issues we might need to worry about with compiling the Windows GNU `.S` files into a binary that's otherwise targeting the MSVC ABI? Like different callee-saved register conventions or whatever? Or is it just the assembly syntax that's different between the two?

Reply:
Syntax only, yes.